extensions/amp-analytics/0.1/events.js

```diff
@@ -380,6 +399,79 @@ export class CustomEventTracker extends EventTracker {
   }
 }
 
+class AmpStoryEventTracker extends EventTracker {
+  /**
+   * @param {!./analytics-root.AnalyticsRoot} root
+   */
+  constructor(root) {
+    super(root);
+
+    /** @private {?Observable<!Event>} */
+    this.sessionObservable_ = new Observable();
+
+    /** @private {?function(!Event)} */
+    this.boundOnSession_ = this.sessionObservable_.fire.bind(
+      this.sessionObservable_
+    );
+
+    Object.keys(StoryAnalyticsEvent).forEach(key => {
+      this.root
+        .getRoot()
+        .addEventListener(StoryAnalyticsEvent[key], this.boundOnSession_);
+    });
+  }
+
+  /** @override */
+  dispose() {
+    const root = this.root.getRoot();
+    Object.keys(StoryAnalyticsEvent).forEach(key => {
+      root.removeEventListener(StoryAnalyticsEvent[key], this.boundOnSession_);
+    });
+    this.boundOnSession_ = null;
+    this.sessionObservable_ = null;
+  }
+
+  /** @override */
+  add(context, eventType, config, listener) {
+    const storySpec = config['storySpec'] || {};
+    const selector = config['selector'] || storySpec['selector'] || ':root';
+    const selectionMethod = config['selectionMethod'] || null;
+    const targetReady = this.root.getElement(
+      context,
+      selector,
+      selectionMethod
+    );
+    const repeat = storySpec['repeat'];
+    const on = config['on'];
+
+    return this.sessionObservable_.add(event => {
+      const {type} = event;
+      const details = /** @type {?JsonObject|undefined} */ (getData(event));
+      const pageId = details['storyPageId'];
+      const pageEvents = details['pageEvents'];
+      const eventsForPage = pageEvents[pageId];
+
+      if (type !== on) {
+        return;
+      }
+
+      if (repeat === false && eventsForPage[type] > 1) {
```

Comment:
Instead of passing the `eventsForPage` as an array, can we pass `eventsForPage[type]` with the event details instead?

Reply:
Not sure I understand, can you elaborate more please?

Reply:
Sorry about the confusion. 
If I understand it correctly, `this.eventsPerPage_` is used to track how much time an event has occurred for each page here in amp-analytics. All we need is a boolean value indicating if the event has occurred before or not. 
Is it possible for the amp-story to pass in that value whenever the analytics event is triggered, along with event.details?

Reply:
SGTM about using a boolean instead of a number, but I think @cvializ preferred (comment: https://github.com/ampproject/amphtml/pull/23030/commits/aa90b03cda559abe6e06a50ce00b84dbbb804807#r297751948) if this logic lived here, instead of `amp-story`.

@cvializ , are you OK with me moving it back to the `story-analytics.js` file?

Reply:
https://github.com/ampproject/amphtml/pull/23030#discussion_r297751948

What @cvializ describes also makes sense to me. But when it comes to how many times an event has been triggered, the story knows the best. 

I figure a special case can help us decide which way to proceed. 
Think about the case where one put two `<amp-analytics>` element on the page, both listens to the story-page-view event for example, and has `repeat` equals to `false`

Say `<amp-analytics id=1>` got laid out very early, and had the listener ready before the first `story-page-view` event arrives. While `<amp-analytics id=2>` only got laid out after the first event arrives. (In theory this won't be an issue, because we buffer all events. But there's still a 10 seconds buffer time) But it still got the second the `story-page-view`. 

What do you think the expected behavior should be.  `<amp-analytics id=2>` still send out the event or not? Personally I'm leaning towards having story keep the count because there could be multiple amp-analytics elements, but only one story that's triggering the events.

Reply:
That's a good point and a use case I didn't think about (having two `amp-analytics` elements). I will change it back so that `amp-story` has the count of events per page - and change it to be a boolean instead of a number. Does this sound good to you? @zhouyx @cvializ

Reply:
👍 That sounds good. I thought `repeat` was a number and not a boolean